components/MarqueeText/src/MarqueeText.cs

```diff
@@ -234,30 +267,29 @@ private bool UpdateAnimation(bool resume = true)
             return false;
         }
 
-        // Swap the start and end to inverse direction for right or upwards
-        if (IsDirectionInverse)
-        {
-            (start, end) = (end, start);
-        }
-
-        // The second segment of text should be hidden if the marquee is not in looping mode
-        _segment2.Visibility = IsLooping ? Visibility.Visible : Visibility.Collapsed;
-
         // Calculate the animation duration by dividing the distance by the speed
         TimeSpan duration = TimeSpan.FromSeconds(distance / Speed);
 
-        // Unbind events from the old storyboard
+        // The second segment of text should be hidden if the marquee is not in looping mode
+        _segment2.Visibility = IsLooping ? Visibility.Visible : Visibility.Collapsed;
+        
+        // Unbind events from the old storyboard and stop it before disposal.
         if (_marqueeStoryboard is not null)
         {
             _marqueeStoryboard.Completed -= StoryBoard_Completed;
+            _marqueeStoryboard?.Stop();
         }
 
         // Create new storyboard and animation
-        _marqueeStoryboard = CreateMarqueeStoryboardAnimation(start, end, duration, targetProperty);
+        _marqueeStoryboard = Behavior switch
```

Comment:
We should consider the ways we can generalize this control so differences like this are handled in the style template instead of internally in the control. Might be a bit of work, but now that things are working under a variety of configurations it would be a good thing to do.